app/src/main/java/net/gsantner/markor/format/todotxt/TodoTxtHighlighter.java

```diff
@@ -38,80 +39,114 @@ public TodoTxtHighlighter(HighlightingEditor hlEditor, Document document) {
     }
 
     @Override
-    protected Editable run(final Editable editable) {
+    protected Spannable run(final Spannable spannable) {
         try {
-            clearSpans(editable);
+            clearSpans(spannable);
 
-            if (editable.length() == 0) {
-                return editable;
+            if (spannable.length() == 0) {
+                return spannable;
             }
 
-            final boolean isDarkBg = _appSettings.isDarkThemeEnabled();
-
             _profiler.start(true, "Todo.Txt Highlighting");
-            generalHighlightRun(editable);
+            generalHighlightRun(spannable);
             _profiler.restart("Paragraph top padding");
-            createParagraphStyleSpanForMatches(editable, LINE_OF_TEXT,
+            createParagraphStyleSpanForMatches(spannable, LINE_OF_TEXT,
                     (matcher, iM) -> new FirstLineTopPaddedParagraphSpan(2f));
 
+            basicTodoTxtHighlights(spannable, false, colors, _appSettings.isDarkThemeEnabled(), _profiler);
+
+            // Paragraph divider
+            _profiler.restart("Paragraph divider");
+            createParagraphStyleSpanForMatches(spannable, LINE_OF_TEXT,
+                    (matcher, iM) -> new HorizontalLineBackgroundParagraphSpan(_hlEditor.getCurrentTextColor(), 0.8f, _hlEditor.getTextSize() / 2f));
+
+            // Fix for paragraph padding and horizontal rule
+            /*
+            nprofiler.restart("Single line fix 1");
+            createRelativeSizeSpanForMatches(spannable, LINESTART, 0.8f);
+            nprofiler.restart("Single line fix 2");
+            createRelativeSizeSpanForMatches(spannable, LINESTART, 1.2f);*/
+            _profiler.restart("Single line fix 1");
+            createRelativeSizeSpanForMatches(spannable, LINESTART, 1.00001f);
+            _profiler.end();
+            _profiler.printProfilingGroup();
+        } catch (Exception ex) {
+            // Ignoring errors
+        }
+
+        return spannable;
+    }
+
+    private static class ProfileCaller {
+        final NanoProfiler _profiler;
+        public ProfileCaller(final NanoProfiler profiler) {
+            _profiler = profiler;
+        }
+        public void restart(final String ... optionalText) {
+            if (_profiler != null) {
+                _profiler.restart(optionalText);
+            }
+        }
+    }
+
+    public static Spannable basicTodoTxtHighlights(
+            final Spannable spannable,
+            final boolean clear,
+            final TodoTxtHighlighterColors colors,
+            final boolean isDarkBg,
+            // Can be null
+            final NanoProfiler profiler
+    ) {
+        try {
+            if (clear) {
+                clearSpans(spannable);
+            }
+
+            if (spannable.length() == 0) {
+                return spannable;
+            }
+
+            ProfileCaller _profiler = new ProfileCaller(profiler);
```

Comment:
Please get rid of this profilecaller stuff. Just pass instance from texteditor.

And for search create instance there and just pass the same always. Profiler anyway not matters there.